### PR TITLE
Avatar Selection Page

### PR DESCRIPTION
**Pull request by ktpss95112**

### CODE REVIEW

File: services/gateway/gateway-service.mjs [hunk 1]
```diff
@@ -205,20 +187,48 @@ class GatewayService {
...
+    // Register all events.
+    socket.on('playerUpdate', (msg) => {
+      this.onPlayerUpdate(socket, PlayerSyncMessage.fromObject(msg));
+      // onPlayerUpdate is async, so returns immediately.
+    });
+
+    // notify the client to start the game after his/her avatar is selected
+    socket.on('avatarSelect', async (msg) => {
+      // initialize the appearance of player
+      socket.playerData.displayName = msg.displayName;
```
Comment (john0312) [inline services/gateway/gateway-service.mjs:221]: Check <14 char?

File: services/gateway/gateway-service.mjs [hunk 2]
```diff
@@ -205,20 +187,48 @@ class GatewayService {
...
+      socket.playerData.displayChar = msg.displayChar;
```
Comment (john0312) [inline services/gateway/gateway-service.mjs:222]: Check it's valid?

File: sites/game-client/client.css [hunk 3]
```diff
@@ -34,6 +39,38 @@ body {
...
 }
 
+#avatar-selection-div {
+  position: absolute;
+  top: 0px;
+  left: 0px;
+  width: 100vw;
+  height: 100vh;
+  box-sizing: border-box;
+  z-index: 100;
```
Comment (john0312) [inline sites/game-client/client.css:49]: Maybe 99 or 90 so error modal will be above it?

File: sites/game-client/game-client.mjs [hunk 4]
```diff
@@ -58,7 +60,7 @@ class GameClient {
       socket.on('unauthorized', (msg) => {
         console.error(`Authorization failed: ${JSON.stringify(msg.data)}`);
         this.mainUI.errorModal.displayError("Authorization failed.", "Please refresh to reconnect.");
-        this.onDisconnect();
+        this.onDisconnect(); //????????? not implemented ????????????
```
Comment (john0312) [inline sites/game-client/game-client.mjs:63]: Open an issue?

**Review by john0312 — APPROVED** [review]

File: services/gateway/gateway-service.mjs (hunk 1 above)
Comment (ktpss95112) [reply services/gateway/gateway-service.mjs:221]: fixed in player.mjs

File: services/gateway/gateway-service.mjs (hunk 2 above)
Comment (ktpss95112) [reply services/gateway/gateway-service.mjs:222]: We don't have graphic asset manager on the server side.
This should be fixed but would involve some code.

File: sites/game-client/client.css (hunk 3 above)
Comment (ktpss95112) [reply sites/game-client/client.css:49]: fixed

File: sites/game-client/game-client.mjs (hunk 4 above)
Comment (ktpss95112) [reply sites/game-client/game-client.mjs:63]: OK.

**Pull request merged**